vmcheck/vmcheck_linux.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vmcheck
+
+import (
+	"encoding/binary"
+	"syscall"
+
+	"github.com/vmware/vmw-guestinfo/bdoor"
+)
+
+// From https://github.com/intel-go/cpuid/blob/master/cpuidlow_amd64.s
+// Get the CPU ID low level leaf values.
+func cpuid_low(arg1, arg2 uint32) (eax, ebx, ecx, edx uint32)
+
+// IsVirtualWorld returns true if running in a VM.
+func IsVirtualWorld() (bool, error) {
+	// Test the HV bit is set
+	if !CPUisVM() {
+		return false, nil
+	}
+
+	// Test if backdoor port is available.
+	if isVM, err := HypervisorPortCheck(); err != nil || !isVM {
+		return isVM, err
+	}
+
+	return true, nil
+}
+
+func HypervisorPortCheck() (bool, error) {
+	if err := syscall.Ioperm(int(bdoor.BackdoorPort), int(bdoor.BackdoorPort)+1, 1); err != nil {
```

Comment:
OK, I think I figured it out.  Although we use the base port, apparently the other ports need to be given perms.  So changing num to 4 seems to work reliably.  I'll test your vmx snippet to disable the backdoor and ping when things are ready.  Thanks for bearing with me on this.  And thanks for looking up that glaring error!